tests: `test_get_text_width_no_locale()` fails on Ubuntu 22.04

### Summary

The unit test ``test_get_text_width_no_locale()`` in test/units/utils/test_display.py fails on Ubuntu 22.04
```
def test_get_text_width_no_locale():
        locale.setlocale(locale.LC_ALL, 'C.UTF-8')
>       pytest.raises(EnvironmentError, get_text_width, '\U000110cd')
E       Failed: DID NOT RAISE <class 'OSError'>

test/units/utils/test_display.py:40: Failed
```

the cause appears to be a new behaviour of the C function `wcwidth()`, which no longer returns -1 for the character used ([U+110CD KAITHI NUMBER SIGN ABOVE](https://www.compart.com/en/unicode/U+110CD), unicode 11.0, June 2018).

```
$ docker run -it --rm public.ecr.aws/n5z0e8q9/ubuntu2004-test python3 -c "import ctypes,ctypes.util; _LIBC = ctypes.cdll.LoadLibrary(ctypes.util.find_library('c')); _LIBC.wcwidth.argtypes = (ctypes.c_wchar,); _LIBC.wcswidth.argtypes = (ctypes.c_wchar_p, ctypes.c_int); _MAX_INT = 2 ** (ctypes.sizeof(ctypes.c_int) * 8 - 1) - 1; print(_LIBC.wcswidth('\\U000110cd', _MAX_INT))"
-1
```

```
$ lsb_release -d
Description:	Ubuntu 22.04.1 LTS
$ python3 -c "import ctypes,ctypes.util; _LIBC = ctypes.cdll.LoadLibrary(ctypes.util.find_library('c')); _LIBC.wcwidth.argtypes = (ctypes.c_wchar,); _LIBC.wcswidth.argtypes = (ctypes.c_wchar_p, ctypes.c_int); _MAX_INT = 2 ** (ctypes.sizeof(ctypes.c_int) * 8 - 1) - 1; print(_LIBC.wcwidth('\\U000110cd'))"
1
```

### Issue Type

Bug Report

### Component Name

ansible-test

### Ansible Version

```console
$ ansible --version
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel"
if you are modifying the Ansible engine, or trying out features under development. This is a rapidly
changing source of code and can become unstable at any point.
ansible [core 2.14.0.dev0] (test_ansible_version-devel 8c6e00df28) last updated 2022/08/25 11:58:21 (GMT +100)
  config file = None
  configured module search path = ['/home/willmerae/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/willmerae/src/ansible/lib/ansible
  ansible collection location = /home/willmerae/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/willmerae/src/ansible/bin/ansible
  python version = 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0] (/usr/bin/python)
  jinja version = 3.0.3
  libyaml = True
```


### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
[WARNING]: You are running the development version of Ansible. You should only
run Ansible from "devel" if you are modifying the Ansible engine, or trying out
features under development. This is a rapidly changing source of code and can
become unstable at any point.
CONFIG_FILE() = None
```


### OS / Environment

Ubuntu 22.04 LTS x64

### Steps to Reproduce

```
$ source hacking/env-setup
$ ansible-test units --venv --python 3.9


### Expected Results

All unit tests pass

### Actual Results

```console
Unit test modules with Python 3.9
============================= test session starts ==============================
platform linux -- Python 3.9.13, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/willmerae/src/ansible, configfile: test/lib/ansible_test/_data/pytest/config/default.ini
plugins: xdist-2.5.0, forked-1.4.0, mock-3.7.0
gw0 I / gw1 I / gw2 I / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I / gw8 I / gw9 I / gw10 I / gw11 I
gw0 [123] / gw1 [123] / gw2 [123] / gw3 [123] / gw4 [123] / gw5 [123] / gw6 [123] / gw7 [123] / gw8 [123] / gw9 [123] / gw10 [123] / gw11 [123]

........................................................................ [ 58%]
...................................................                      [100%]
- generated xml file: /home/willmerae/src/ansible/test/results/junit/python3.9-modules-units.xml -
============================= slowest 25 durations =============================
0.07s call     test/units/modules/test_hostname.py::TestHostname::test_stategy_get_never_writes_in_check_mode
0.03s call     test/units/modules/test_async_wrapper.py::TestAsyncWrapper::test_run_module
0.03s call     test/units/modules/test_iptables.py::TestIptables::test_log_level
0.02s setup    test/units/modules/test_service.py::test_sunos_service_start
0.01s call     test/units/modules/test_known_hosts.py::KnownHostsDiffTestCase::test_sanity_check
0.01s call     test/units/modules/test_copy.py::test_split_pre_existing_dir_working_dir_exists[dir1-expected0]
0.01s setup    test/units/modules/test_copy.py::test_split_pre_existing_dir_one_level_exists[dir1/dir2/-expected3]
0.01s call     test/units/modules/test_iptables.py::TestIptables::test_iprange
0.01s call     test/units/modules/test_iptables.py::TestIptables::test_tcp_flags
0.01s setup    test/units/modules/test_copy.py::test_split_pre_existing_dir_one_level_exists[/dir1/-expected5]
0.01s call     test/units/modules/test_iptables.py::TestIptables::test_chain_creation
0.01s setup    test/units/modules/test_copy.py::test_good_symbolic_modes[32768-a+X-0]
0.01s setup    test/units/modules/test_copy.py::test_good_symbolic_modes[16384-u+rwx-448]
0.01s call     test/units/modules/test_copy.py::test_split_pre_existing_dir_working_dir_exists[dir1/dir2-expected2]
0.01s setup    test/units/modules/test_pip.py::test_failure_when_pip_absent[patch_ansible_module0]
0.01s setup    test/units/modules/test_copy.py::test_good_symbolic_modes[16895-g-rwx-455]
0.01s setup    test/units/modules/test_copy.py::test_good_symbolic_modes[16384-g=rwx-56]
0.01s call     test/units/modules/test_iptables.py::TestIptables::test_chain_deletion_check_mode
0.01s call     test/units/modules/test_iptables.py::TestIptables::test_without_required_parameters
0.01s setup    test/units/modules/test_copy.py::test_good_symbolic_modes[16895-a-X-438]
0.01s setup    test/units/modules/test_copy.py::test_good_symbolic_modes[16384-g+rwx-56]
0.01s setup    test/units/modules/test_copy.py::test_good_symbolic_modes[16384-u=rwx,g=rwx,o=rwx-511]
0.01s call     test/units/modules/test_iptables.py::TestIptables::test_insert_rule_with_wait
0.01s setup    test/units/modules/test_copy.py::test_good_symbolic_modes[16895-a-rwx-0]
0.01s setup    test/units/modules/test_copy.py::test_good_symbolic_modes[16384-u+rwx,g+rwx,o+rwx-511]
============================= 123 passed in 1.40s ==============================
Unit test module_utils with Python 3.9
============================= test session starts ==============================
platform linux -- Python 3.9.13, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/willmerae/src/ansible, configfile: test/lib/ansible_test/_data/pytest/config/default.ini
plugins: xdist-2.5.0, forked-1.4.0, mock-3.7.0
gw0 I / gw1 I / gw2 I / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I / gw8 I / gw9 I / gw10 I / gw11 I
gw0 [1673] / gw1 [1673] / gw2 [1673] / gw3 [1673] / gw4 [1673] / gw5 [1673] / gw6 [1673] / gw7 [1673] / gw8 [1673] / gw9 [1673] / gw10 [1673] / gw11 [1673]

........................................................................ [  4%]
........................................................................ [  8%]
.......................s....s....s..s..s....s...ss...s.s.s.......s.....s [ 12%]
....s................................................................... [ 17%]
........................................................................ [ 21%]
........................................................................ [ 25%]
........................................................................ [ 30%]
........................................................................ [ 34%]
......................................................................... [ 38%]
........................................................................ [ 43%]
........................................................................ [ 47%]
........................................................................ [ 51%]
........................................................................ [ 56%]
........................................................................ [ 60%]
........................................................................ [ 64%]
........................................................................ [ 68%]
........................................................................ [ 73%]
........................................................................ [ 77%]
........................................................................ [ 81%]
.s......s.........................................s....s.s.s............ [ 86%]
..................s..................................................... [ 90%]
........................................................................ [ 94%]
........................................................................ [ 99%]
................                                                         [100%]
- generated xml file: /home/willmerae/src/ansible/test/results/junit/python3.9-module_utils-units.xml -
============================= slowest 25 durations =============================
3.01s call     test/units/module_utils/facts/test_timeout.py::test_implicit_file_overridden_succeeds
2.01s call     test/units/module_utils/facts/test_timeout.py::test_explicit_timeout
2.01s call     test/units/module_utils/facts/test_timeout.py::test_explicit_succeeds
2.01s call     test/units/module_utils/facts/test_timeout.py::test_implicit_file_overridden_timesout
1.02s call     test/units/module_utils/facts/test_timeout.py::test_implicit_file_default_timesout
1.01s call     test/units/module_utils/facts/test_timeout.py::test_implicit_file_default_succeeds
1.01s call     test/units/module_utils/facts/test_timeout.py::test_timeout_raises_timeout
1.01s call     test/units/module_utils/facts/test_timeout.py::test_exception_not_caught_by_called_code
1.00s call     test/units/module_utils/facts/test_timeout.py::test_timeout_raises_timeout_integration_test[stdin0]
1.00s call     test/units/module_utils/test_api.py::TestRetryWithDelaysAndCondition::test_retry_exception
1.00s call     test/units/module_utils/test_api.py::TestRateLimit::test_ratelimit
0.22s call     test/units/module_utils/facts/test_facts.py::TestFactsLinuxHardwareGetMountFacts::test_get_mount_facts
0.22s call     test/units/module_utils/facts/hardware/test_linux.py::TestFactsLinuxHardwareGetMountFacts::test_get_mount_facts
0.14s call     test/units/module_utils/facts/test_ansible_collector.py::TestPkgMgrOSTreeFacts::test_is_rhel_edge_ostree
0.14s call     test/units/module_utils/facts/test_ansible_collector.py::TestExceptionCollectedFacts::test_expected_facts
0.14s call     test/units/module_utils/facts/test_ansible_collector.py::TestInPlace::test1
0.13s call     test/units/module_utils/facts/test_ansible_collector.py::TestPkgMgrFacts::test_expected_facts
0.13s call     test/units/module_utils/facts/test_ansible_collector.py::TestPkgMgrFacts::test_not_expected_facts
0.13s call     test/units/module_utils/facts/test_ansible_collector.py::TestPkgMgrOSTreeFacts::test_is_fedora_ostree
0.13s call     test/units/module_utils/facts/test_ansible_collector.py::TestPkgMgrFacts::test_basics
0.13s call     test/units/module_utils/facts/test_ansible_collector.py::TestPkgMgrOSTreeFacts::test_not_expected_facts
0.12s call     test/units/module_utils/facts/test_ansible_collector.py::TestCollectedFacts::test_basics
0.12s call     test/units/module_utils/facts/test_ansible_collector.py::TestExceptionCollectedFacts::test_basics
0.12s call     test/units/module_utils/facts/test_ansible_collector.py::TestCollectedFacts::test_expected_facts
0.12s call     test/units/module_utils/facts/test_ansible_collector.py::TestExceptionCollectedFacts::test_not_expected_facts
=========================== short test summary info ============================
SKIPPED [1] test/units/module_utils/basic/test_imports.py:84: literal_eval is available in every version of Python3
SKIPPED [5] test/units/module_utils/basic/test_log.py:38: python systemd bindings not installed
SKIPPED [5] test/units/module_utils/basic/test_log.py:130: python systemd bindings not installed
SKIPPED [1] test/units/module_utils/basic/test_log.py:139: python systemd bindings not installed
SKIPPED [2] test/units/module_utils/basic/test_log.py:113: python systemd bindings not installed
SKIPPED [1] test/units/module_utils/facts/test_collectors.py:370: faulty test
SKIPPED [1] test/units/module_utils/facts/test_collectors.py:396: faulty test
SKIPPED [1] test/units/module_utils/facts/test_facts.py:46: This platform (DragonFly) does not have a fact_class.
SKIPPED [3] test/units/module_utils/facts/test_facts.py:64: This test class needs to be updated to specify collector_class
SKIPPED [1] test/units/module_utils/facts/test_facts.py:54: This platform (DragonFly) does not have a fact_class.
====================== 1652 passed, 21 skipped in 12.17s =======================
Unit test controller with Python 3.9
============================= test session starts ==============================
platform linux -- Python 3.9.13, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/willmerae/src/ansible, configfile: test/lib/ansible_test/_data/pytest/config/default.ini
plugins: xdist-2.5.0, forked-1.4.0, mock-3.7.0
gw0 I / gw1 I / gw2 I / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I / gw8 I / gw9 I / gw10 I / gw11 I
gw0 [1871] / gw1 [1871] / gw2 [1871] / gw3 [1871] / gw4 [1871] / gw5 [1871] / gw6 [1871] / gw7 [1871] / gw8 [1871] / gw9 [1871] / gw10 [1871] / gw11 [1871]

........................................................................ [  3%]
........................................................................ [  7%]
........................................................................ [ 11%]
........................................................................ [ 15%]
........................................................................ [ 19%]
........................................................................ [ 23%]
........................................................................ [ 26%]
........................................................................ [ 30%]
........................................................................ [ 34%]
........................................................................ [ 38%]
.........................................s.............................. [ 42%]
........................................................................ [ 46%]
........................................................................ [ 50%]
........................................................................ [ 53%]
........................................................................ [ 57%]
........................................................................ [ 61%]
........................................................................ [ 65%]
........................................................................ [ 69%]
........................................................................ [ 73%]
........................................................................ [ 76%]
...................s....s.....s....s......s.....s....................... [ 80%]
........................................................................ [ 84%]
..........F............................................................. [ 88%]
........................................................................ [ 92%]
........................................................................ [ 96%]
.................................s.....................................  [100%]
=================================== FAILURES ===================================
________________________ test_get_text_width_no_locale _________________________
[gw7] linux -- Python 3.9.13 /home/willmerae/src/ansible/test/results/.tmp/delegation/python3.9/bin/python
def test_get_text_width_no_locale():
        locale.setlocale(locale.LC_ALL, 'C.UTF-8')
>       pytest.raises(EnvironmentError, get_text_width, '\U000110cd')
E       Failed: DID NOT RAISE <class 'OSError'>

test/units/utils/test_display.py:40: Failed
- generated xml file: /home/willmerae/src/ansible/test/results/junit/python3.9-controller-units.xml -
============================= slowest 25 durations =============================
2.16s call     test/units/utils/test_encrypt.py::test_encrypt_with_ident
2.11s call     test/units/executor/test_task_executor.py::TestTaskExecutor::test_task_executor_run_loop
1.15s call     test/units/plugins/lookup/test_password.py::TestLookupModuleWithPasslibWrappedAlgo::test_encrypt_wrapped_crypt_algo
1.13s call     test/units/utils/test_encrypt.py::test_encrypt_default_rounds
1.01s call     test/units/galaxy/test_api.py::test_wait_import_task_timeout[https://galaxy.server.com/api-v2-Token-token_ins0-1234-https://galaxy.server.com/api/v2/collection-imports/1234/]
1.00s call     test/units/galaxy/test_api.py::test_wait_import_task_timeout[https://galaxy.server.com/api/automation-hub-v3-Bearer-token_ins1-1234-https://galaxy.server.com/api/automation-hub/v3/imports/collections/1234/]
0.77s call     test/units/cli/test_galaxy.py::TestGalaxy::test_execute_remove
0.73s call     test/units/cli/test_galaxy.py::TestGalaxy::test_run
0.61s call     test/units/cli/test_adhoc.py::test_simple_command
0.60s call     test/units/utils/test_encrypt.py::test_do_encrypt_passlib
0.52s call     test/units/parsing/yaml/test_loader.py::TestAnsibleLoaderVault::test_embedded_vault
0.45s call     test/units/utils/test_encrypt.py::test_passlib_bcrypt_salt
0.44s call     test/units/utils/display/test_logger.py::test_logger
0.38s call     test/units/cli/test_galaxy.py::test_install_implicit_role_with_collections[\ncollections:\n- namespace.name\nroles:\n- namespace.name\n]
0.37s call     test/units/utils/collection_loader/test_collection_loader.py::test_import_from_collection
0.36s call     test/units/utils/collection_loader/test_collection_loader.py::test_collpkg_loader_load_module
0.33s call     test/units/utils/collection_loader/test_collection_loader.py::test_collection_get_data
0.32s call     test/units/utils/collection_loader/test_collection_loader.py::test_bogus_imports
0.32s call     test/units/cli/test_galaxy.py::test_parse_requirements_with_collection_source[\ncollections:\n- name: namespace.collection\n- name: namespace2.collection2\n  source: https://galaxy-dev.ansible.com/\n- name: namespace3.collection3\n  source: server\n]
0.32s call     test/units/cli/test_galaxy.py::test_parse_requirements_with_roles_and_collections[\nroles:\n- username.role_name\n- src: username2.role_name2\n- src: ssh://github.com/user/repo\n  scm: git\n\ncollections:\n- namespace.collection2\n]
0.30s call     test/units/cli/test_galaxy.py::test_parse_requirements_roles_with_include_missing[\n- username.role\n- include: missing.yml\n]
0.29s call     test/units/cli/test_galaxy.py::test_parse_requirements_roles_with_include[\n- username.included_role\n- src: https://github.com/user/repo\n]
0.25s call     test/units/executor/test_play_iterator.py::TestPlayIterator::test_play_iterator
0.23s call     test/units/executor/test_play_iterator.py::TestPlayIterator::test_play_iterator_nested_blocks
0.21s call     test/units/ansible_test/ci/test_azp.py::test_auth
=========================== short test summary info ============================
SKIPPED [1] test/units/parsing/vault/test_vault.py:783: This test is not ready yet
SKIPPED [1] test/units/plugins/strategy/test_strategy.py:121: Temporarily disabled due to fragile tests that need rewritten
SKIPPED [1] test/units/plugins/strategy/test_strategy.py:43: Temporarily disabled due to fragile tests that need rewritten
SKIPPED [1] test/units/plugins/strategy/test_strategy.py:425: Temporarily disabled due to fragile tests that need rewritten
SKIPPED [1] test/units/plugins/strategy/test_strategy.py:218: Temporarily disabled due to fragile tests that need rewritten
SKIPPED [1] test/units/plugins/strategy/test_strategy.py:175: Temporarily disabled due to fragile tests that need rewritten
SKIPPED [1] test/units/plugins/strategy/test_strategy.py:69: Temporarily disabled due to fragile tests that need rewritten
SKIPPED [1] test/units/vars/test_module_response_deepcopy.py:40: No current support for this situation
FAILED test/units/utils/test_display.py::test_get_text_width_no_locale
================== 1 failed, 1862 passed, 8 skipped in 9.20s ===================
FATAL: Command "pytest --forked -r a -n auto --color yes -p no:cacheprovider -c /home/willmerae/src/ansible/test/lib/ansible_test/_data/pytest/config/default.ini --junit-xml /home/willmerae/src/ansible/test/results/junit/python3.9-controller-units.xml --strict-markers --rootdir /home/willmerae/src/ansible --durations=25 test/units/_vendor/test_vendor.py test/units/ansible_test/ci/test_azp.py test/units/cli/arguments/test_optparse_helpers.py test/units/cli/galaxy/test_collection_extract_tar.py test/units/cli/galaxy/test_display_collection.py test/units/cli/galaxy/test_display_header.py test/units/cli/galaxy/test_display_role.py test/units/cli/galaxy/test_execute_list.py test/units/cli/galaxy/test_execute_list_collection.py test/units/cli/galaxy/test_get_collection_widths.py test/units/cli/test_adhoc.py test/units/cli/test_cli.py test/units/cli/test_console.py test/units/cli/test_doc.py test/units/cli/test_galaxy.py test/units/cli/test_playbook.py test/units/cli/test_vault.py test/units/config/manager/test_find_ini_config_file.py test/units/config/test_manager.py test/units/errors/test_errors.py test/units/executor/module_common/test_modify_module.py test/units/executor/module_common/test_module_common.py test/units/executor/module_common/test_recursive_finder.py test/units/executor/test_interpreter_discovery.py test/units/executor/test_play_iterator.py test/units/executor/test_playbook_executor.py test/units/executor/test_task_executor.py test/units/executor/test_task_queue_manager_callbacks.py test/units/executor/test_task_result.py test/units/galaxy/test_api.py test/units/galaxy/test_collection.py test/units/galaxy/test_collection_install.py test/units/galaxy/test_role_install.py test/units/galaxy/test_role_requirements.py test/units/galaxy/test_token.py test/units/galaxy/test_user_agent.py test/units/inventory/test_group.py test/units/inventory/test_host.py test/units/parsing/test_ajson.py test/units/parsing/test_dataloader.py test/units/parsing/test_mod_args.py test/units/parsing/test_splitter.py test/units/parsing/test_unquote.py test/units/parsing/utils/test_addresses.py test/units/parsing/utils/test_jsonify.py test/units/parsing/utils/test_yaml.py test/units/parsing/vault/test_vault.py test/units/parsing/vault/test_vault_editor.py test/units/parsing/yaml/test_constructor.py test/units/parsing/yaml/test_dumper.py test/units/parsing/yaml/test_loader.py test/units/parsing/yaml/test_objects.py test/units/playbook/role/test_include_role.py test/units/playbook/role/test_role.py test/units/playbook/test_attribute.py test/units/playbook/test_base.py test/units/playbook/test_block.py test/units/playbook/test_collectionsearch.py test/units/playbook/test_conditional.py test/units/playbook/test_helpers.py test/units/playbook/test_included_file.py test/units/playbook/test_play.py test/units/playbook/test_play_context.py test/units/playbook/test_playbook.py test/units/playbook/test_taggable.py test/units/playbook/test_task.py test/units/plugins/action/test_action.py test/units/plugins/action/test_gather_facts.py test/units/plugins/action/test_pause.py test/units/plugins/action/test_raw.py test/units/plugins/become/test_su.py test/units/plugins/become/test_sudo.py test/units/plugins/cache/test_cache.py test/units/plugins/callback/test_callback.py test/units/plugins/connection/test_connection.py test/units/plugins/connection/test_local.py test/units/plugins/connection/test_paramiko.py test/units/plugins/connection/test_psrp.py test/units/plugins/connection/test_ssh.py test/units/plugins/connection/test_winrm.py test/units/plugins/filter/test_core.py test/units/plugins/filter/test_mathstuff.py test/units/plugins/inventory/test_constructed.py test/units/plugins/inventory/test_inventory.py test/units/plugins/inventory/test_script.py test/units/plugins/lookup/test_env.py test/units/plugins/lookup/test_ini.py test/units/plugins/lookup/test_password.py test/units/plugins/lookup/test_url.py test/units/plugins/shell/test_cmd.py test/units/plugins/shell/test_powershell.py test/units/plugins/strategy/test_linear.py test/units/plugins/strategy/test_strategy.py test/units/plugins/test_plugins.py test/units/regex/test_invalid_var_names.py test/units/template/test_native_concat.py test/units/template/test_templar.py test/units/template/test_template_utilities.py test/units/template/test_vars.py test/units/test_constants.py test/units/test_context.py test/units/test_no_tty.py test/units/utils/collection_loader/test_collection_loader.py test/units/utils/display/test_broken_cowsay.py test/units/utils/display/test_display.py test/units/utils/display/test_logger.py test/units/utils/display/test_warning.py test/units/utils/test_cleanup_tmp_file.py test/units/utils/test_context_objects.py test/units/utils/test_display.py test/units/utils/test_encrypt.py test/units/utils/test_helpers.py test/units/utils/test_isidentifier.py test/units/utils/test_plugin_docs.py test/units/utils/test_shlex.py test/units/utils/test_unsafe_proxy.py test/units/utils/test_vars.py test/units/utils/test_version.py test/units/vars/test_module_response_deepcopy.py test/units/vars/test_variable_manager.py" returned exit status 1.
FATAL: Command "/usr/bin/env ANSIBLE_TEST_CONTENT_ROOT=/home/willmerae/src/ansible PYTHONPATH=/tmp/ansible-test-qhlhnur_ /home/willmerae/src/ansible/test/results/.tmp/delegation/python3.9/bin/python /home/willmerae/src/ansible/bin/ansible-test units --containers '{}' --truncate 105 --color yes --host-path test/results/.tmp/host-ah1wk7w7 --metadata test/results/.tmp/metadata-np2gu58c.json" returned exit status 1.
```


### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

--- a/test/units/utils/test_display.py
+++ b/test/units/utils/test_display.py
@@ -5,7 +5,6 @@
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
-import ctypes.util
 import locale
 import sys
 import unicodedata

--- a/test/units/utils/test_display.py
+++ b/test/units/utils/test_display.py
@@ -5,13 +5,32 @@
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
+import ctypes.util
 import locale
+import sys
+import unicodedata
 from unittest.mock import MagicMock
 
 import pytest
 
-from ansible.utils.display import Display, get_text_width
+from ansible.utils.display import _LIBC, _MAX_INT, Display, get_text_width
 from ansible.utils.multiprocessing import context as multiprocessing_context
+
+
+@pytest.fixture
+def problematic_wcswidth_chars():
+    problematic = []
+    try:
+        locale.setlocale(locale.LC_ALL, 'C.UTF-8')
+    except Exception:
+        return problematic
+
+    candidates = set(chr(c) for c in range(sys.maxunicode) if unicodedata.category(chr(c)) == 'Cf')
+    for c in candidates:
+        if _LIBC.wcswidth(c, _MAX_INT) == -1:
+            problematic.append(c)
+
+    return problematic
 
 
 def test_get_text_width():
@@ -35,9 +54,11 @@ def test_get_text_width():
     pytest.raises(TypeError, get_text_width, b'four')
 
 
-def test_get_text_width_no_locale():
+def test_get_text_width_no_locale(problematic_wcswidth_chars):
+    if not problematic_wcswidth_chars:
+        pytest.skip("No problmatic wcswidth chars")
     locale.setlocale(locale.LC_ALL, 'C.UTF-8')
-    pytest.raises(EnvironmentError, get_text_width, '\U000110cd')
+    pytest.raises(EnvironmentError, get_text_width, problematic_wcswidth_chars[0])
 
 
 def test_Display_banner_get_text_width(monkeypatch):

